server.go

```diff
@@ -323,6 +326,19 @@ func (srv *Server) SetOption(option Option) error {
 	return option(srv)
 }
 
+func (srv *Server) CloseConnectionsForUser(user string) {
```

Comment:
This is too specific to your use-case. A better way to generalize this would be to make connections exported or a hook to get all connections or something, so the specific implementations can do whatever they want with all the connections.